src/user-search/client.ts

```diff
@@ -0,0 +1,294 @@
+        } else {
+            this.cache = cache
+        }
+        if (client === undefined) {
+            this.client = new Client(address)
+        } else {
+            this.client = client
+        }
+    }
+
+    public async getExtendedAccount(account: string): Promise<ExtendedAccount> {
+        const [result] = await this.call('condenser_api', 'get_accounts', [[account]])
+        return result
+    }
+
+    public async getFollowCount(account: string): Promise<FollowCountReturn> {
+        return await this.call('condenser_api', 'get_follow_count', [account])
+    }
+
+    public async* followGen(method: string, account: string, followType: FollowType, start = '') {
+        let pageCount = 0
+        while (true) {
+            pageCount += 1
+            const results = await this.call('condenser_api', method, [
+                account,
+                start,
+                followType,
+                this.pageSize
+            ])
+            const lastResult = results[results.length - 1]
+            yield* results
+            logger.info(`p:${pageCount} start:${start} last:${lastResult} length:${results.length}`)
+            if (start === lastResult  || results.length < this.pageSize) {
+                break
+            }
+            if (method === 'get_followers') {
+                start = lastResult['follower']
+            } else {
+                start = lastResult['following']
+            }
+        }
+    }
+
+    public async getFollowers(account: string, start = 0, limit = 1000): Promise<FollowReturn[]> {
+        const followType = FollowType.blog
+        const followers: any[] = []
+        for await (const results of this.followGen('get_followers', account, followType )) {
+            followers.push(results)
+        }
+        return followers
+    }
+
+    public async getIgnored(account: string, start = 0, limit = 1000): Promise<FollowReturn[]> {
+        const followType = FollowType.ignore
+        const followers: any[] = []
+        for await (const results of this.followGen('get_followers', account, followType )) {
+            followers.push(results)
+        }
+        return followers
+    }
+
+    public async getFollowing(account: string, start = 0, limit = 1000): Promise<FollowReturn[]> {
+        const followType = FollowType.blog
+        const followers: any[] = []
+        for await (const results of this.followGen('get_following', account, followType )) {
+            followers.push(results)
+        }
+        return followers
+    }
+
+    public async* accountHistoryGenerator(account: string) {
+        let pointer: number = -1
+        let pageCount = 0
+        let accountHistoryLength: number
+        let totalPages: number = 1
+        while (true) {
+            pageCount += 1
+            const resultsPage = await this.call(
+                'condenser_api',
+                'get_account_history',
+                [account, pointer, this.pageSize])
```

Comment:
It does use pageSize: https://github.com/steemit/conveyor/blob/user-search-api/src/user-search/client.ts#L143 but my implemention is unclear because of letovers from a previous approach. I'll fix that